components/prism-cooklang.js

```diff
@@ -0,0 +1,152 @@
+(function (Prism) {
+
+	var single_token_suffix = /(?:(?!\s)[\d$+<=a-zA-Z\x80-\uFFFF])+/.source;
+	var multi_token_infix = /[^{}@#]+/.source;
+	var multi_token_suffix = /\{[^}#@]*\}/.source;
+
+	var multi_token = multi_token_infix + multi_token_suffix;
+
+	var timer_units = /(?:h|hours|hrs|m|min|minutes)/.source;
+
+	var amount_group_impl = {
+		pattern: new RegExp(/\{/.source
+			+ '(?:(?:'
+			+ /(?:[^{}|*%]+\*?)/.source // optional serving scale
+			+ '|'
+			+ /(?:(?:[^{}|*%]+\|)+[^{}|*%]+)/.source // serving alternatives
+			+ ')' + /(?:%[^{}|*%]+)?/.source //optional unit
+			+ ')?'
+			+ /\}/.source
+		),
+		inside: {
+			'amount': {
+				pattern: /([\{|])[^{}|*%]+/,
+				lookbehind: true,
+				alias: 'number',
+			},
+			'unit': {
+				pattern: /(%)[^}]+/,
+				lookbehind: true,
+				alias: 'symbol',
+			},
+			'servings-scaler': {
+				pattern: /\*/,
+				alias: 'operator',
+			},
+			'servings-alternative-separator': {
+				pattern: /\|/,
+				alias: 'operator',
+			},
+			'unit-separator': {
+				pattern: /(?:%|(\*)%)/,
+				lookbehind: true,
+				alias: 'operator',
+			},
+			'punctuation': /[{}]/,
+		}
+	};
+
+
+	Prism.languages.cooklang = {
+		'comment': {
+			// [- comment -]
+			// -- comment
+			pattern: /\[-[\s\S]*?-\]|--.*/,
+			greedy: true,
+		},
+		'meta': { // >> key: value
+			pattern: />>.*:.*/,
+			inside: {
+				'property': { // key:
+					pattern: /(>>\s*)[^\s:](?:[^:]*[^\s:])?/,
+					lookbehind: true,
+				}
+			}
+		},
+		'cookware-group': { // #...{...}, #...
+			pattern: new RegExp('#(?:'
+				+ multi_token
+				+ '|'
+				+ single_token_suffix
+				+ ')'
+			),
+			inside: {
+				'cookware': {
+					pattern: new RegExp('(^#)(?:'
+						+ multi_token_infix
+						+ ')'
+					),
+					lookbehind: true,
+					alias: 'variable',
+				},
+				'cookware-keyword': {
+					pattern: /^#/,
+					alias: 'keyword',
+				},
+				'quantity-group': {
+					pattern: new RegExp(/\{[^{}@#]*\}/),
+					inside: {
+						'quantity': {
+							pattern: new RegExp(/(^\{)/.source + multi_token_infix),
+							lookbehind: true,
+							alias: 'number',
+						},
+						'punctuation': /[{}]/,
+					}
+				}
+			},
+		},
+		'ingredient-group': { // @...{...}, @...
+			pattern: new RegExp('@(?:'
+				+ multi_token
+				+ '|'
+				+ single_token_suffix
+				+ ')'),
+			inside: {
+				'ingredient': {
+					pattern: new RegExp('(^@)(?:'
+						+ multi_token_infix
+						+ ')'),
+					lookbehind: true,
+					alias: 'variable',
+				},
+				'ingredient-keyword': {
+					pattern: /^@/,
+					alias: 'keyword',
+				},
+				'amount-group': amount_group_impl,
+			}
+		},
+		'timer-group': { // ~timer{...}
+			// eslint-disable-next-line regexp/sort-alternatives
+			pattern: new RegExp(/~[^{}]*\{\d+%/.source + timer_units + /\}/.source),
```

Comment:
1. The EBNF also allows spaces between the amount/duration tokens. It might be better to be a bit more lenient here and to allow pretty much anything inside the braces. Being more lenient also has the benefit that we will give good syntax highlighting while typing.
2. I think `~[^{}]*` is a little too general. We should probably exclude a few of Cooklang's special characters.
```suggestion
			pattern: /~(?!\s)[^@#~{}]*\{[^{}]*\}/,
```